src/jimgw/transforms.py

```diff
@@ -0,0 +1,177 @@
+from abc import ABC, abstractmethod
+from typing import Callable
+
+import jax
+import jax.numpy as jnp
+from chex import assert_rank
+from jaxtyping import Float
+
+
+class Transform(ABC):
+    """
+    Base class for transform.
+
+    The idea of transform should be used on distribtuion,
+    """
+
+    name_mapping: tuple[list[str], list[str]]
+    transform_func: Callable[[dict[str, Float]], dict[str, Float]]
+
+    def __init__(
+        self,
+        name_mapping: tuple[list[str], list[str]],
+    ):
+        self.name_mapping = name_mapping
+
+    def __call__(self, x: dict[str, Float]) -> tuple[dict[str, Float], Float]:
+        return self.transform(x)
+
+    @abstractmethod
+    def transform(self, x: dict[str, Float]) -> tuple[dict[str, Float], Float]:
```

Comment:
What should one do with the Jacobian for generic N to M transformations?

Reply:
We actually need the determinant of the Jacobian instead of the Jacobian matrix itself, which is not defined in the case of N to M transform. For N to M transform, we should explicitly prevent its use as a transform for a distribution. An N to M transform can only be used as a function (Also most likely only in one direction).